src/Microsoft.AspNetCore.Razor/Tokenizer/Symbols/SymbolExtensions.cs

```diff
@@ -13,7 +13,21 @@ public static class SymbolExtensions
     {
         public static LocationTagged<string> GetContent(this SpanBuilder span)
         {
-            return GetContent(span, e => e);
+            var symbols = span.Symbols;
+            if (symbols.Count > 0)
+            {
+                var text = new StringBuilder();
```

Comment:
If this is deemed to be expensive we can make an alternate call path that takes in a pooled builder.